### PR TITLE
Migrate TestKillDifferentUserContainer to api test

### PR DESCRIPTION
**Pull request by yongtang**
This fix migrates TestKillDifferentUserContainer to api test

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: integration/container/kill_test.go [hunk 1]
```diff
@@ -175,3 +175,21 @@ func TestKillStoppedContainerAPIPre120(t *testing.T) {
 	err = client.ContainerKill(ctx, c.ID, "SIGKILL")
 	require.NoError(t, err)
 }
+
+func TestKillDifferentUserContainer(t *testing.T) {
+	// TODO Windows: Windows does not yet support -u (Feb 2016).
+	skip.If(t, testEnv.OSType != "linux")
```
Comment (vdemeester) [inline integration/container/kill_test.go]: Should we print "Windows does not support -u" in the skip ?

Comment (vdemeester) [reply integration/container/kill_test.go]: Well the message should be different as we're not using the `cli` at all

**Review by vdemeester — APPROVED** [review]
LGTM 🌵

File: integration/container/kill_test.go (hunk 1 above)
Comment (yongtang) [reply integration/container/kill_test.go]: @vdemeester Updated. Thanks.

**Review by dnephin — APPROVED** [review]
LGTM

**Pull request merged**